### PR TITLE
Fix our MSB3270 warnings

### PR DESCRIPTION
**Pull request by jaredpar**
This has two changes:

- Elevates MSB3270 and MSB3277 to errors in our build
- Fixes the existing MSB3270 warnings in our code

SDK behavior change link: https://github.com/dotnet/sdk/issues/3495

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure @nguerrera for review

File: src/Compilers/Server/VBCSCompilerTests/VBCSCompiler.UnitTests.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <TargetFrameworks>$(RoslynPortableTargetFrameworks)</TargetFrameworks>
     <RuntimeIdentifiers>$(RoslynPortableRuntimeIdentifiers)</RuntimeIdentifiers>
+    <PlatformTarget>x86</PlatformTarget>
```
Comment (nguerrera) [inline src/Compilers/Server/VBCSCompilerTests/VBCSCompiler.UnitTests.csproj]: You should set VBCSCompiler explciitly to AnyCPU not this to x86. 

I saw there's a place where it is set to x86 for bootstrap so you'll want to not override that case to AnyCPU

**Comment by jaredpar** [comment]
Hold off reviewing, this change is wrong.

**Comment by jaredpar** [comment]
This is ready now again.

**Comment by nguerrera** [comment]
I still see the wrong change.

**Comment by jaredpar** [comment]
@nguerrera fixed ... it's been a great first week back for me.

**Comment by jaredpar** [comment]
Closing as this change is incomlpete. PR https://github.com/dotnet/roslyn/pull/37646 has the preferred approach.

**Pull request closed without merging**